app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepositoryImpl.kt

```diff
@@ -65,87 +74,106 @@ class AgendaRepositoryImpl @Inject constructor(
         }
     }
 
-    override suspend fun updateLocalAgendaDayFromRemote(dateTime: ZonedDateTime) {
-        withContext(Dispatchers.IO) {
-            try {
-                // Get fresh data
-                val result = agendaApi.getAgenda(dateTime)
-
-                if (result.isSuccess) {
-                    val agenda = result.getOrNull()
-
-                    eventRepository.clearEventsForDayLocally(dateTime)
-                    // Insert fresh data locally
-                    agenda?.events?.forEach { event ->
-                        val result2 =
-                            eventRepository.upsertEventLocally(event.toDomain())
-                        if (result2 is ResultUiText.Error) {
-                            throw IllegalStateException(result2.message.asStrOrNull())
-                        }
+    override suspend fun updateLocalAgendaDayFromRemote(dateTime: ZonedDateTime): ResultUiText<Unit> {
+        try {
+            // Get fresh data
+            val result = agendaApi.getAgenda(dateTime)
+
+            if (result.isSuccess) {
+                val agenda = result.getOrNull()
+
+                eventRepository.clearEventsForDayLocally(dateTime)
+                // Insert fresh data locally
+                agenda?.events?.forEach { event ->
+                    val result2 =
+                        eventRepository.upsertEventLocally(event.toDomain())
+                    if (result2 is ResultUiText.Error) {
+                        throw IllegalStateException(result2.message.asStrOrNull())
                     }
+                }
 
-                    taskRepository.clearTasksForDayLocally(dateTime)
-                    // Insert fresh data locally
-                    agenda?.tasks?.forEach { task ->
-                        val result2 =
-                            taskRepository.upsertTaskLocally(task.toDomain())
-                        if (result2 is ResultUiText.Error) {
-                            throw IllegalStateException(result2.message.asStrOrNull())
-                        }
+                taskRepository.clearTasksForDayLocally(dateTime)
+                // Insert fresh data locally
+                agenda?.tasks?.forEach { task ->
+                    val result2 =
+                        taskRepository.upsertTaskLocally(task.toDomain())
```

Comment:
I assume this is asynchronous? Then you could execute all these calls in parallel

Reply:
ok will refactor